src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1254,27 +1255,31 @@ private Void emitTemplatizedType(TemplatizedType type, boolean extendingInstance
         emit(inImplementsExtendsPosition ? ">" : "[]");
         return null;
       }
-      switch (type.getDisplayName()) {
-        // Arguments<?> and NodeList<?> in es3 externs are correspondingly
-        // IArguments and NodeList interfaces (not-parametrized) in lib.d.ts.
-        case "Arguments":
-          emit("IArguments");
-          return null;
-        case "NodeList":
-          emit("NodeList");
-          return null;
-        case "MessageEvent":
-          emit("MessageEvent");
-          return null;
-        case "IThenable":
-          templateTypeName = "PromiseLike";
-          break;
-        case "IArrayLike":
-          templateTypeName = "ArrayLike";
-          break;
-        default:
-          break;
+
+      if (!opts.allowPlatformExterns) {
```

Comment:
+1 let me know if you need help with the testing infra.

It would be quite a large output, because if you don't pass externs the behavior is indistingushable, but if you do the output will contain all externs. Maybe you can think of a way to mock out the externs to only a few files instead of all platform externs. If not, I will be ok with a largish output for now.